components/calendar/src/cal/hijri.rs

```diff
@@ -525,7 +528,7 @@ impl HijriSimulatedLocation {
             }
             other => {
                 debug_assert!(
-                    false,
+                    !WELL_BEHAVED_ASTRONOMICAL_RANGE.contains(&start_day),
```

Comment:
Thought: I've often been annoyed by having to write `debug_assert!(false, ...)` instead of something like `debug_panic!(...)`, but it makes this type of refactor cleaner